src/toil/toilState.py

```diff
@@ -40,51 +42,56 @@ def __init__(self, jobStore, rootJob, jobCache=None):
         the JobStore, and is not preserved.
 
         :param toil.jobStores.abstractJobStore.AbstractJobStore jobStore
+
         :param toil.job.JobDescription rootJob
+
+        :param Dict[str, JobDescription] jobCache
         """
 
         # Maps from successor (child or follow-on) jobStoreID to predecessor jobStoreID
-        self.successorJobStoreIDToPredecessorJobs = {}
+        self.successorJobStoreIDToPredecessorJobs: Dict[str, List[JobDescription]] = {}
 
         # Hash of jobStoreIDs to counts of numbers of successors issued.
         # There are no entries for jobs without successors in this map.
-        self.successorCounts = {}
+        self.successorCounts: Dict[str, int] = {}
 
         # This is a hash of service jobs, referenced by jobStoreID, to their predecessor job
-        self.serviceJobStoreIDToPredecessorJob = {}
+        self.serviceJobStoreIDToPredecessorJob: Dict[str, JobDescription] = {}
 
         # Hash of jobStoreIDs mapping to dict from service ID to service host JobDescription for issued services
-        self.servicesIssued = {}
+        self.servicesIssued: Dict[str, JobDescription] = {}
 
         # Jobs that are ready to be processed.
         # Stored as a dict from job store ID to a pair of (job, result status),
         # where a status other than 0 indicates that an error occurred when
         # running the job.
-        self.updatedJobs = {}
+        self.updatedJobs: Dict[str, Tuple[JobDescription, int]] = {}
 
         # The set of totally failed jobs - this needs to be filtered at the
         # end to remove jobs that were removed by checkpoints
-        self.totalFailedJobs = set()
+        self.totalFailedJobs: Set[JobDescription] = set()
 
         # Jobs (as jobStoreIDs) with successors that have totally failed
-        self.hasFailedSuccessors = set()
+        self.hasFailedSuccessors: Set[str] = set()
 
         # The set of successors of failed jobs as a set of jobStoreIds
-        self.failedSuccessors = set()
+        self.failedSuccessors: Set[str] = set()
 
         # Set of jobs that have multiple predecessors that have one or more predecessors
         # finished, but not all of them. This acts as a cache for these jobs.
         # Stored as hash from jobStoreIDs to JobDescriptions
-        self.jobsToBeScheduledWithMultiplePredecessors = {}
+        self.jobsToBeScheduledWithMultiplePredecessors: Dict[str, JobDescription] = {}
 
-        ##Algorithm to build this information
+        # Algorithm to build this information
         self._buildToilState(rootJob, jobStore, jobCache)
 
 
-    def allJobDescriptions(self):
+    def allJobDescriptions(self) -> Iterator[JobDescription]:
         """
         Returns an iterator over all JobDescription objects referenced by the
         ToilState, with some possibly being visited multiple times.
+
+        :rtype: Iterator[JobDescription]
```

Comment:
I don't think you should add rtype lines; we probably want to drop all of them in favor of the real type hints instead.